lib/hash.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+def deeply_sort_hash(object)
+  return object unless object.is_a?(Hash)
+
+  hash = {}
+  object.each { |k, v| hash[k] = deeply_sort_hash(v) }
+  sorted = hash.sort { |a, b| a[0].to_s <=> b[0].to_s }
```

Comment:
I think this should work too
```suggestion
  hash = object.transform_values { |v| deeply_sorted_hash(v) }.sort_by(&:to_s)
```